engine/schema/src/com/cloud/dc/dao/ClusterDaoImpl.java

```diff
@@ -260,4 +268,41 @@ public boolean remove(Long id) {
         sc.setParameters("dataCenterId", zoneId);
         return customSearch(sc, null);
     }
+
+    @Override
+    public boolean computeWhetherClusterSupportsResigning(long clusterId) {
+        ClusterVO cluster = findById(clusterId);
+
+        if (cluster == null || cluster.getAllocationState() != Grouping.AllocationState.Enabled) {
+            return false;
+        }
+
+        List<HostVO> hosts = hostDao.findByClusterId(clusterId);
+
+        if (hosts == null) {
+            return false;
+        }
+
+        for (HostVO host : hosts) {
+            if (host == null) {
+                return false;
+            }
+
+            DetailVO hostDetail = hostDetailsDao.findDetail(host.getId(), "supportsResign");
+
+            if (hostDetail == null) {
+                return false;
+            }
+
+            String value = hostDetail.getValue();
+
+            Boolean booleanValue = Boolean.valueOf(value);
+
+            if (booleanValue == false) {
+                return false;
+            }
+        }
+
+        return true;
+    }
```

Comment:
The DAO calls in the ``for`` loop are concerning from a performance perspective (both for the speed of this call and the load placed on the database).  Ideally, it would be calculated in a single join.  Would it be possible to craft a query across the details for all hosts in a cluster where the detail key = "supportsResign" and the value = "true"?